core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1188,11 +1188,9 @@ class ReplicaManager(val config: KafkaConfig,
                           correlationId: Int,
                           responseMap: mutable.Map[TopicPartition, Errors],
                           highWatermarkCheckpoints: OffsetCheckpoints): Set[Partition] = {
-    partitionState.keys.foreach { partition =>
-      stateChangeLogger.trace(s"Handling LeaderAndIsr request correlationId $correlationId from " +
-        s"controller $controllerId epoch $controllerEpoch starting the become-leader transition for " +
-        s"partition ${partition.topicPartition}")
-    }
+    stateChangeLogger.info(s"Handling LeaderAndIsr request correlationId $correlationId from " +
+      s"controller $controllerId epoch $controllerEpoch starting the become-leader transition for " +
+      s"${partitionState.size} partitions ${partitionState.keys.map(p => p.topicPartition.toString).mkString(", ")}")
```

Comment:
If we have multiple partitions, logging them in a single log line is hard to read. So, perhaps the current approach of one line per partition is better? Ditto in a few other places. Once we make a decision here, it would be useful to use that consistently in all places.